azurerm/internal/services/datafactory/data_factory_integration_runtime_self_hosted.go

```diff
@@ -0,0 +1,272 @@
+	client := meta.(*clients.Client).DataFactory.IntegrationRuntimesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	factoryName := id.Path["factories"]
+	name := id.Path["integrationruntimes"]
+
+	resp, err := client.Get(ctx, resourceGroup, factoryName, name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error retrieving Data Factory Self-Hosted Integration Runtime %q (Resource Group %q, Data Factory %q): %+v", name, resourceGroup, factoryName, err)
+	}
+
+	d.Set("name", name)
+	d.Set("data_factory_name", factoryName)
+	d.Set("resource_group_name", resourceGroup)
+
+	selfHostedIntegrationRuntime, convertSuccess := resp.Properties.AsSelfHostedIntegrationRuntime()
+
+	if !convertSuccess {
+		return fmt.Errorf("Error converting integration runtime to Self-Hosted integration runtime %q (Resource Group %q, Data Factory %q)", name, resourceGroup, factoryName)
+	}
+
+	if selfHostedIntegrationRuntime.Description != nil {
+		d.Set("description", selfHostedIntegrationRuntime.Description)
+	}
+
+	if props := selfHostedIntegrationRuntime.SelfHostedIntegrationRuntimeTypeProperties; props != nil {
+		// LinkedInfo BasicLinkedIntegrationRuntimeType
+		if linkedInfo := props.LinkedInfo; linkedInfo != nil {
+			rbacAuthorization, _ := linkedInfo.AsLinkedIntegrationRuntimeRbacAuthorization()
+			if rbacAuthorization != nil {
+				if err := d.Set("rbac_authorization", schema.NewSet(resourceArmDataFactoryIntegrationRuntimeSelfHostedRbacAuthorizationHash, flattenAzureRmDataFactoryIntegrationRuntimeSelfHostedTypePropertiesRbacAuthorization(rbacAuthorization))); err != nil {
+					return fmt.Errorf("Error setting `rbac_authorization`: %#v", err)
+				}
+			}
+		}
+		return nil
+	}
+
+	respKey, errKey := client.ListAuthKeys(ctx, resourceGroup, factoryName, name)
+	if errKey != nil {
+		if utils.ResponseWasNotFound(respKey.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error retrieving Data Factory Self-Hosted Integration Runtime %q Auth Keys (Resource Group %q, Data Factory %q): %+v", name, resourceGroup, factoryName, errKey)
+	}
+
+	d.Set("auth_key_1", respKey.AuthKey1)
+	d.Set("auth_key_1", respKey.AuthKey2)
+
+	return nil
+}
+
+func resourceArmDataFactoryIntegrationRuntimeSelfHostedDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataFactory.IntegrationRuntimesClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	factoryName := id.Path["factories"]
+	name := id.Path["integrationruntimes"]
```

Comment:
Could we use a resource specific parse id function here like can be seen in the analysis package: parse.AnalysisServicesServerID